worker/sshserver/listener.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"net"
+	"sync"
+)
+
+// sshServerListener is required to prevent a race condition
+// that can occur in tests.
+//
+// The SSH server tracks the listeners in use
+// but if the server's close() method executes
+// before we reach a safe point in the Serve() method
+// then the server's map of listeners will be empty.
+// A safe point to indicate the server is ready is
+// right before we start accepting connections.
+// Accept() will return with error if the underlying
+// listener is already closed.
```

Comment:
This doesn't seem like a test-only issue.

What happens if we never get an accept, does this then block until it does get one? Something doesn't seem quite right...